plugins/hydra_ray_launcher/hydra_plugins/hydra_ray_launcher/conf/__init__.py

```diff
@@ -0,0 +1,211 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import Any, Dict, List, Optional
+
+from hydra.core.config_store import ConfigStore
+from hydra.types import ObjectConf
+from omegaconf import MISSING
+
+
+@dataclass
+class RayInitConf(Dict[str, Any]):
+    """ parameters for ray.init()"""
+
+    pass
+
+
+@dataclass
+class RayRemoteConf(Dict[str, Any]):
+    """ parameters for @ray.remote() """
+
+    num_cpus: Optional[int] = 1
+    num_gpus: Optional[int] = 0
+
+
+@dataclass
+class RayConf:
+    ray_init_cfg: RayInitConf = field(default_factory=RayInitConf)
+    ray_remote_cfg: RayRemoteConf = field(default_factory=RayRemoteConf)
```

Comment:
Once you are no longer extending Dict you can simplify:
```suggestion
    ray_init_cfg: RayInitConf = RayInitConf()
    ray_remote_cfg: RayRemoteConf = RayRemoteConf()
```

Reply:
I've read the Structured Config doc a few times but I don't have a clear idea on what's the best way to address my use case here other than extending ```Dict[str, Any]```.

There's (at least) 3 fields I need to support for RayInitConf
1.  num_cpus
2. num_gpus 
3. address

But I don't always need all 3 of them present, for example I only need to set ```address``` when running in remote mode. But if I define the class as the following and not passing in ```address```

```

@dataclass
class RayInitConf:
    """ parameters for ray.init()"""
    num_cpus = MISSING
    num_gpus = MISSING
    address = MISSING
```
the config will be interpreted as ```address=None``` and pass to ```ray.init()``` when I just need the field to not show up at all.


A user may also want to pass other params into ```ray.init``` other than the ones I listed.

Reply:
resolving. Modified the dataclass per our offline discussion.